### PR TITLE
Autostake

### PR DESCRIPTION
**Pull request by Dellybro**

### CODE REVIEW

File: contracts/Autostake.sol [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        uint256 _amountOut,
+        uint256 _deadline,
+        uint256 _days
+    ) external returns (uint256[] memory amounts) {
+        require(
+            allowedTokens[_token] == true,
+            'AUTOSTAKER: This token is not allowed to be used on this contract'
+        );
+
+        amounts = _swapTokensForTokens(_token, _amount, _amountOut, _deadline);
```
Comment (Mvii7) [inline contracts/Autostake.sol:56]: Should return the single amount used, and refactored in the future if more than 1 is needed

**Comment by Mvii7** [comment]
Tests failing

File: contracts/Autostake.sol [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    Token Contract: Burner (AKA Minter)
+ */
+
+contract Autostake is Initializable, Migrateable, Manageable {
+    event Stake(address from, address tokenIn, uint256 amountOut);
+
+    address staking;
+    address uniswap;
+    address axion;
+    mapping(address => bool) public allowedTokens;
```
Comment (Mvii7) [inline contracts/Autostake.sol:24]: Why do we need this? 
We can make it look exclusive to BAT in the UI

Comment (Dellybro) [reply contracts/Autostake.sol:26]: We've learned from the past "exclusive" to the frontend is not how contracts should work. We need a check in the backend.

File: contracts/Autostake.sol [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import './abstracts/Manageable.sol';
+import './interfaces/IToken.sol';
+
+/** Launch
+    Roles Needed -
+    Staking Contract: External Staker Role
+    Token Contract: Burner (AKA Minter)
+ */
+
+contract Autostake is Initializable, Migrateable, Manageable {
```
[the remark this answers is not in the document]
Comment (Dellybro) [reply contracts/Autostake.sol:19]: Autostaking axion bought with other tokens.

**Pull request merged**